### ISSUE TITLE
Merchant Hosted payments

### ISSUE THREAD

**Issue by samthejarvis** [swipestripe/silverstripe-swipestripe#30]
Would I be right in assuming that SwipeStripe doesn't support the merchant hosted payment flow?

I've tried it with the https://github.com/frankmullenger/silverstripe-gsoc-payment-test (merchant hosted) and it just seems to redirect straight to the completed order page, without the credit card input/check stage.

**Comment by Zauberfisch** [comment]
I am currently building a site using SwipeStripe and a payment gateway, I had to implement quiet a bit of handling myself to get it working.
I have already spoken with @frankmullenger about the payment handling, and there are some inconsistencies and problems that need to be fixed. But we have not arrived at a solution yet.

But I can't really speak about the module you are trying to use, maybe @frankmullenger is able to point you in the right direction

**Comment by frankmullenger** [comment]
@samthejarvis  sorry for the delayed reply on this. The merchant hosting should really work so I'm not sure what to say here. Have you tried with a gateway or just the demo/test code?

**Comment by samthejarvis** [comment]
Not at all, we're using a gateway hosted PSP now so there's no hurry. I'll have another look over this, probably did something wrong.

Gateway hosted works perfectly, this is about merchant hosted gateways where payment details are collected on the merchant's site. I can't see any code in the swipestripe checkout flow referring to serving a form that collects these details (credit card number etc) so I assumed it wasn't implemented. I tried creating my own merchant hosted payment gateway but couldn't manage to implement the credit card details collection step.

**Comment by frankmullenger** [comment]
@samthejarvis Oh ok, from memory the form is wrapped up in the Payment module and it _should_ just work :-)

**Comment by martinwolfaardt** [comment]
I would so appreciate it if anyone could point me in the right direction on this issue. The getFormFields method in my extension of the PaymentProcessor_MerchantHosted class is never called - there doesn't seem to be an intermediate stage where a form is served that handles the capture of credit card details etc. I am confused!

**Comment by quinnmerio** [comment]
I am having a similar issue. None of the payment modules (tried Stripe and PayPal) will render fields to collect credit card details. The processing happens right away and generates errors due to the missing data.

**Comment by frankmullenger** [comment]
@quinnmerio sorry, it's been a long time since I looked at this module. It may be easier to integrate Omnipay than debug this I'm afraid.
